Lib/doctest.py

```diff
@@ -1140,6 +1140,8 @@ def _find_lineno(self, obj, source_lines):
             obj = obj.fget
         if inspect.isfunction(obj) and getattr(obj, '__doc__', None):
             # We don't use `docstring` var here, because `obj` can be changed.
+            while hasattr(obj, "__wrapped__"):
+                obj = obj.__wrapped__
```

Comment:
`doctest` already has a heavy dependency on the `inspect` module -- consider using [`inspect.unwrap()`](https://docs.python.org/3/library/inspect.html#inspect.unwrap) here

Reply:
`inspect.unwrap` fits perfectly, thanks! Made the change.